File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, Field


class UserNotificationResponse(BaseModel):
    id: int
    recipient_user_id: int
    category: str
    title: str
    body: str = ""
    action_url: str | None = None
    action_label: str | None = None
    is_read: bool
    dedupe_key: str | None = None
    payload_json: dict = Field(default_factory=dict)
    created_at: datetime | None = None
    read_at: datetime | None = None


class NotificationListResponse(BaseModel):
    unread_count: int = 0
    items: list[UserNotificationResponse] = Field(default_factory=list)


class CodexResetWatchdogHistoryItemResponse(BaseModel):
    item_id: str
    text: str
    url: str | None = None
    author: str | None = None
    published_at: str | None = None
    translation_zh: str | None = None
    reset_status: str | None = None
    reset_evidence: str | None = None
    archived_at: str | None = None


class CodexResetWatchdogHistoryResponse(BaseModel):
    source_handle: str
    max_items: int = 30
    archived_count: int = 0
    last_success_at: str | None = None
    items: list[CodexResetWatchdogHistoryItemResponse] = Field(default_factory=list)

File: frontend/src/types/notification.ts
export type NotificationCategory =
  | 'strategy_received'
  | 'collection_required'
  | 'collection_ready'
  | 'codex_reset_watchdog'

export interface UserNotification {
  id: number
  recipient_user_id: number
  category: NotificationCategory
  title: string
  body: string
  action_url: string | null
  action_label: string | null
  is_read: boolean
  dedupe_key: string | null
  payload_json: Record<string, unknown>
  created_at: string | null
  read_at: string | null
}

export interface NotificationListResponse {
  unread_count: number
  items: UserNotification[]
}

export type CodexResetStatus = 'scheduled' | 'completed' | 'possible'

export interface CodexResetWatchdogHistoryItem {
  item_id: string
  text: string
  url: string | null
  author: string | null
  published_at: string | null
  translation_zh: string | null
  reset_status: CodexResetStatus | null
  reset_evidence: string | null
  archived_at: string | null
}

export interface CodexResetWatchdogHistory {
  source_handle: string
  max_items: number
  archived_count: number
  last_success_at: string | null
  items: CodexResetWatchdogHistoryItem[]
}
